File: PYTHON/lista1/zad1.py
# Rafał Leja 340879
# zadanie 1 lista 1 "vat"
# 6.10.2023

zakupy = [0.1, 0.1, 0.1, 0.1, 0.1, 10]

def vat_faktura(lista):
  return sum(lista) * 0.23

def vat_paragon(lista):
  vat = 0.0;
  for x in lista:
    vat += x*0.23
  return vat

print(vat_faktura(zakupy))
print(vat_paragon(zakupy))
print(vat_faktura(zakupy) == vat_paragon(zakupy))

File: MIA/lista1/zad1.c
/*
* Rafał Leja 340879
* MIA 2025
* Zadanie A, Lista 1
* 07.10.2025
*/
#include <stdlib.h>
#include <stdio.h>

int main(void)
{
  int input;
  scanf("%d", &input);

  int banknoty [] = { 100, 20, 10, 5, 1 };
  int nominal = 0;
  int ilosc = 0;
  while (input > 0) {
    int akt_ilosc = input / banknoty[nominal];
    if (akt_ilosc > 0){
      ilosc += akt_ilosc;
      input -= akt_ilosc * banknoty[nominal];
    } 

    nominal++;
  }

  printf("%d\n", ilosc);
  return EXIT_SUCCESS;
}
